quic/s2n-quic-core/src/datagram/default.rs

```diff
@@ -316,6 +345,11 @@ impl Sender {
     ///
     /// If the datagram queue is at capacity the oldest datagram will be popped
     /// off the queue and returned to make space for the newest datagram.
+    ///
+    /// # Return value
+    /// - `Ok(None)` if the datagram was enqueued for sending
+    /// - `Ok(Some(Bytes))` if the queue is at capacity this will be the oldest datagram on the queue
+    /// - `Err(DatagramError)` if some error occurred
```

Comment:
`if self.queue.len() == self.capacity {`

- what if the len doesnt match the capacity exactly?



```
  oldest = self.queue.pop_front();
...
        self.queue.push_back(datagram);
```
seems like we only `pop_front` one Bytes. Is that guarnteed to be the same size as the one we are `push_back`. If not then we could be growing the VecDeque slowly over time.

Reply:
This function pops old datagrams off the queue if the queue reaches the user-requested capacity. So, if the length is smaller than the capacity we don't need to do anything. Also, pop_front will pop more than just one byte. You can see that functionality [on this line](https://github.com/aws/s2n-quic/blob/main/quic/s2n-quic-core/src/datagram/default.rs#L544) of the test.